### PR TITLE
KOGITO-1467 Invalid enpoint url in process instance list

### PR DESCRIPTION
**Pull request by AjayJagan**
- added serviceUrl support for process management.

### CODE REVIEW

File: packages/management-console/server/MockData/types.js [hunk 1]
```diff
@@ -135,6 +136,7 @@ module.exports = typeDefs = gql`
     error: ProcessInstanceErrorArgument
     nodes: NodeInstanceArgument
     endpoint: StringArgument
+    serviceUrl: StringArgument
```
Comment (cristianonicolai) [inline packages/management-console/server/MockData/types.js]: not exposed to filter on

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 2]
```diff
@@ -53,6 +54,7 @@ interface IProcessInstance {
   start: string;
   state: string;
   addons: string[];
+  serviceUrl: string | null;
   endpoint: string;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: remove endpoint?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 3]
```diff
@@ -37,6 +37,7 @@ import {
 } from '@patternfly/react-icons';
 import ErrorPopover from '../../Atoms/ErrorPopoverComponent/ErrorPopoverComponent';
 import ProcessBulkModalComponent from '../../Atoms/ProcessBulkModalComponent/ProcessBulkModalComponent';
+import DisablePopupWithCheckboxComponent from '../DiablePopupComponent/DisablePopupComponent';
 
 /* tslint:disable:no-string-literal */
 export interface IProcessInstanceError {
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: use TypeScript ProcessInstance?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 4]
```diff
@@ -244,15 +245,15 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
...
       axios
         .delete(
-          `${processInstanceData.endpoint}/management/processes/${processId}/instances/${processInstanceId}`
+          `${processInstanceData.serviceUrl}/management/processes/${processId}/instances/${processInstanceId}`
         )
         .then(() => {
           setModalTitle('Process aborted');
           setModalContent(`${processId} - process execution has been aborted.`);
           setTitleType('success');
-          processInstanceData.state = 'ABORTED';
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: I think you should keep this, only update the state if the request succeds

File: packages/management-console/src/graphql/queries.tsx [hunk 5]
```diff
@@ -24,6 +24,7 @@ const GET_PROCESS_INSTANCES = gql`
       lastUpdate
       addons
       endpoint
```
Comment (cristianonicolai) [inline packages/management-console/src/graphql/queries.tsx]: remove endpoint?

File: packages/management-console/src/graphql/queries.tsx [hunk 6]
```diff
@@ -47,6 +48,7 @@ const GET_CHILD_INSTANCES = gql`
       start
       lastUpdate
       endpoint
```
Comment (cristianonicolai) [inline packages/management-console/src/graphql/queries.tsx]: remove endpoint?

File: packages/management-console/src/graphql/types.tsx [hunk 7]
```diff
@@ -518,27 +514,27 @@ export type __Field = {
   deprecationReason?: Maybe<Scalars['String']>;
 };
 
-/** 
+/**
```
Comment (cristianonicolai) [inline packages/management-console/src/graphql/types.tsx]: @AjayJagan there are quite a lot of changes in this file, just concerned if we dont break something. Could it be just the formatting? Is tslint running when you commit?

**Review by cristianonicolai — COMMENTED** [review]
@AjayJagan the endpoint url link inthe process instance list also needs to be updated. Link needs to point ot serviceUrl value if present. Disabled if serviceUrl is null and show same message as the checkbox.

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 8]
```diff
@@ -499,6 +505,23 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
...
+                  >
+                    Endpoint
+                    {
+                      <ExternalLinkAltIcon
+                        className="pf-u-ml-xs"
+                        color="grey"
+                      />
+                    }
+                  </Button>
                 )}
```
Comment (AjayJagan) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: @srambach - if a button is disabled , does the Tooltip work? I tried and it is not working for me. Thoughts on this please :)

File: packages/management-console/src/components/Molecules/DiablePopupComponent/DisablePopupComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Tooltip, DataListCheck } from '@patternfly/react-core';
+
+interface IOwnProps {
+  processInstanceData: any;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DiablePopupComponent/DisablePopupComponent.tsx]: ProcessInstance  type?

Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DiablePopupComponent/DisablePopupComponent.tsx]: aah sorry forgot :(

Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DiablePopupComponent/DisablePopupComponent.tsx]: done

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 10]
```diff
@@ -38,10 +41,13 @@ import {
 } from '@patternfly/react-icons';
 import ErrorPopover from '../../Atoms/ErrorPopoverComponent/ErrorPopoverComponent';
 import ProcessBulkModalComponent from '../../Atoms/ProcessBulkModalComponent/ProcessBulkModalComponent';
-interface IOwnProps {
+import DisablePopupWithCheckboxComponent from '../DiablePopupComponent/DisablePopupComponent';
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: drop _Component_ from the name?

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 11]
```diff
@@ -38,10 +41,13 @@ import {
...
+
+/* tslint:disable:no-string-literal */
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: is this still needed?

**Review by cristianonicolai — COMMENTED** [review]
@AjayJagan please review https://github.com/kiegroup/kogito-apps/pull/115/files#diff-060b086ad50199c262ec23c03a91359fR506, to:
`href={`${processInstanceData.serviceUrl}`}`

**Review by cristianonicolai — APPROVED** [review]

**Review by nmirasch — APPROVED** [review]

**Comment by Sara4994** [comment]
@AjayJagan  all good. just one thing, i see a key warning when i expand to see child process instances. you need add key for expansion list in DataListItemComponent.jsx. Also i see some conflicts when i take a pull.

Screenshot from 2020-03-27 10-35-22

**Comment by AjayJagan** [comment]
> @AjayJagan all good. just one thing, i see a key warning when i expand to see child process instances. you need add key for expansion list in DataListItemComponent.jsx. Also i see some conflicts when i take a pull.
> 
> Screenshot from 2020-03-27 10-35-22

sure, resolved it now

**Review by Sara4994 — APPROVED** [review]

**Pull request merged**